StevensCAD Senior Design Work

ajohnso4 opened the pull request:
Pull request for the work done for the senior design team.

kaiqichee committed:
added curve files

added .hh, .cc, and test file for curve. updated corresponding CMakeLists. Doesn't work right now, dependent on Transformation (implemented differently than in java)

ajohnso4 committed:
Added Cylinder

ajohnso4 committed:
Added cylinder

kaiqichee committed:
vector header and curve code update

from meeting on 10/20

Commit:
commit

ajohnso4 committed:
new commit

kaiqichee committed:
updated curve tests

kaiqichee committed:
began bezier.hh

kaiqichee committed:
added bezier .hh and .cc

filled in functions from .java files, still need to compile and add Matrix3D

kaiqichee committed:
update cMakeList.txt to include all files added

ajohnso4 committed:
Cylinder changes

ajohnso4 committed:
Merge branch 'curve_kaiqichee' of github.com:ajohnso4/GrailGUI into ajohnson-3dshapes

Commit:
created triangle, rectangle, and circle with tests

ajohnso4 committed:
Update to cylinder

Commit:
additions

ajohnso4 committed:
cylinder changes

kaiqichee committed:
compiled bezier

start point seems off, but compiles and runs

kaiqichee committed:
compiled curve

was having compilation error without main in curve.cc

kaiqichee committed:
check and compile circle, add functions to vec3d

added cross product and normalize to vec3d
*circle.cc for-loop only prints out 6 points, change increment to get more points

kaiqichee committed:
compile curve and test points

draws curve, but is perpendicular to given points

kaiqichee committed:
change Circle extends and import typo

ajohnso4 committed:
rectangular prism created

kaiqichee committed:
add grail window

*doesnt draw

Commit:
update

ajohnso4 committed:
Updated cylinder and rectangularprism

ajohnso4 committed:
fixes

Commit:
circle update

kaiqichee committed:
get drawing on curve gui and create helix

*lines are still too straight for curve, looks disjoint

alikolenovic committed:
Bug fixes

ajohnso4 committed:
started sphere

ajohnso4 committed:
Merge branch 'ajohnson-3dshapes' of github.com:ajohnso4/GrailGUI into ajohnson-3dshapes

ajohnso4 committed:
sphere update

kaiqichee committed:
got helix drawing with z=0

Commit:
gear update

Commit:
gear update

Commit:
gear

Commit:
gear

Commit:
gear

Commit:
gear update

Commit:
gear update

kaiqichee committed:
change camera to view helix

kaiqichee committed:
add transformation to CAD directory and increase points to draw

alikolenovic committed:
commit

ajohnso4 committed:
transformation code added

ajohnso4 committed:
pulled stuff

ajohnso4 committed:
added joe2D

ajohnso4 committed:
saving transform code and test

kaiqichee committed:
bezier path changes

linkage error :(

ahuston-0 committed:
Fix for multiple linker issues

Issues:
  Previously, attempting to compile this would lead to a linker error
due to src/CAD/ not being included in the compile step. This was fixed
relatively quickly, but exposed several multiple-definition issues with
CAD/Vec3d.hh. These issues seemed unusual, but were fixed as soon as the
functions in question were moved into the class definition. Issues like
this are usually due to a .cc/.cpp file being included as if it were a
header file, but that does not seem to be an issue here.

Changelog:

- src/CMakeLists.txt: CAD subdirectory was never added before attempting
  to add the grail-CAD variable to the build path. I'm not sure if this
  ever would have worked.
- BezierPath.cc: CAD/Vec3d.hh was included in BezierPath.hh, so the
  redundant include was removed
- BezierPath.hh: changed Vec3d include from "./Vec3d.hh" to
  "CAD/Vec3d.hh" for clarity purposes
- CAD/CMakeLists.txt: BezierPath was never being compiled, instead
  CMake would have tried to compile a non-existent file called
  Bezier.cc. Given that this directory was never being compiled though,
  it didn't emit any errors.
- Curve.hh: See BezierPath.hh, the same fix was applied.
- Vec3d.hh: A lot of the visible changes are just from running the
  formatter on it. However, the most apparent changes are from moving
  the functions outside the class definition (dot, cross, distance,
  operator<<, operator>>) into the class as friend functions. For
  whatever reason, leaving them outside the function definition is
  causing multiple definition errors. Moving them into the class
  definition seems to have resolved this.
- CurveTest.cc: Mostly style changes with the includes.

ahuston-0 committed:
fixing change with include in BezierPath.hh

kaiqichee committed:
Merge pull request #2 from ahuston-0/bezier_kaiqichee

Linker errors in bezier_kaiqichee

kaiqichee committed:
notes

kaiqichee committed:
Merge branch 'bezier_kaiqichee' of https://github.com/ajohnso4/GrailGUI into bezier_kaiqichee

kaiqichee committed:
find duplicate getPoints() bug

draws bezier curve without the extra line now :)

katechen75 committed:
Update README.md

Commit:
changes

kaiqichee committed:
Merge branch 'main' into bezier_kaiqichee

kaiqichee committed:
Merge pull request #3 from ajohnso4/bezier_kaiqichee

added paths info

kaiqichee committed:
Merge branch 'main' into joe2D

kaiqichee committed:
Merge pull request #4 from ajohnso4/joe2D

Joe2 d

ahuston-0 committed:
fixed issues breaking compile

ahuston-0 committed:
re-enable CurveTest compile

kaiqichee committed:
Merge pull request #5 from ahuston-0/extrusions-kq

Minor Bug Fixes and CurveTest

kaiqichee committed:
started extrusions

kaiqichee committed:
Merge branch 'extrusions-kq' of https://github.com/ajohnso4/GrailGUI into extrusions-kq

ajohnso4 committed:
try this for cylinder

alikolenovic committed:
commit

alikolenovic committed:
test Prism should work but not

alikolenovic committed:
Delete NewCylinder.cc

alikolenovic committed:
Delete NewCylinder.hh

alikolenovic committed:
Delete testNewCylinder.cc

ajohnso4 committed:
bezier fixes

ajohnso4 committed:
update

ajohnso4 committed:
3D shape added

ajohnso4 committed:
Sphere rendering

kaiqichee committed:
created facets, working on extruding along path

ajohnso4 committed:
Rectangular Prism working, with some simple fixes. Sphere working

ajohnso4 committed:
Rectangular prism bug found and dealt with

ajohnso4 committed:
Update README.md

ajohnso4 committed:
Update README.md

ajohnso4 committed:
Update README.md

ajohnso4 committed:
Merge pull request #6 from ajohnso4/ajohnson-3dshapes

Add 3d shapes completed

kaiqichee committed:
changed logic so facet draws along a path

doesn't over draw anymore, added square extrusion on path with 4 vectors

kaiqichee committed:
update CMAKELIST to build helix and extrude along helix

doesn't look completely 3d because of shading

kaiqichee committed:
added extrusion class (.hh and .cc)

a lot more generalized, not manually made in the curve test file anymore

kaiqichee committed:
add tests for each class

added tests for: bezier, extrusion, helix, facet, vec3d, line, circle

kaiqichee committed:
added a test with multiple connected paths

ajohnso4 committed:
Added transform, edited rectangular prism notation

kaiqichee committed:
Merge branch 'main' into extrusions-kq

kaiqichee committed:
Merge pull request #7 from ajohnso4/extrusions-kq

Extrusions kq

kaiqichee committed:
fixed helix bug

alikolenovic committed:
cylinder

alikolenovic committed:
Merge branch 'ajohnson-3dshapes' of https://github.com/ajohnso4/GrailGUI into ajohnson-3dshapes

alikolenovic committed:
Cylinder with Transform

ajohnso4 committed:
finished 3d

ajohnso4 committed:
Merge branch 'main' into ajohnson-3dshapes

ajohnso4 committed:
Merge pull request #8 from ajohnso4/ajohnson-3dshapes

Ajohnson 3dshapes

ahuston-0 committed:
Restoring settings.json to original state

We generally prefer having error squiggles on for debugging purposes, and try not to clog the git history with changes to the file association list.

ahuston-0 commented on test/CircleTest.cc:
```diff
@@ -0,0 +1,43 @@
+    Vec3D a(1,1,1);
+    Vec3D b(2,3,4);
+
+    Circle* firstCircle = new Circle(1, a, b, c, baseGraphStyle);
```
For some reason, these seem to miss drawing the last segment of the circle, so the circle almost connects but doesn't. I'll have to check out Circle.cc and see if there's a fix.

ahuston-0 replied:
This may be a need to switch to a different GL drawing method, or we may need to copy the first element to the last.

ahuston-0 commented on test/Vec3DTest.cc:
```diff
@@ -0,0 +1,36 @@
+};
+
+int main(int argc, char *argv[]) {
+  return GLWin::init(new Vec3DTest(), 1000, 1000);
```
Love this, but this may actually be okay to get rid of the GLWin and just do the Vec3D math.

ahuston-0 commented on test/testCylinder.cc:
```diff
@@ -0,0 +1,42 @@
+        // Cylinder* cyl = ;
+        // c->addLayer(cyl);
+        Transformation* rotate = new Transformation(c, s, new Cylinder(c, s, 4, 60, 60, 0, 1, 200));
+        rotate->rotate(pi/3, 0, 2, 0);
```
This seems to have the same issue as CircleTest where it just does not seem to connect the last point.

ahuston-0 replied:
See CircleTest's comments

ahuston-0 commented on test/testGear.cc:
```diff
@@ -0,0 +1,17 @@
+  void init() {
+    MainCanvas* c = currentTab()->getMainCanvas();
+    const Style* s = getDefaultStyle();
+    Gear* circ = c->addLayer(new Gear(c, 500.0, 500.0, 100.0, 100.0, 6.0, 100.0, 100.0, s));    
```
This has a funky glitch where the first and last point are connecting to (0,0). This is definitely not the fault of anyone on StevensCAD team though, because this is an issue that we ended up fixing in a different branch a while ago. Nice work.

ahuston-0 replied:
Correction, this is not using multishape, so its something we'll have to diagnose later.

ahuston-0 commented on test/testRectangularPrism.cc:
```diff
@@ -0,0 +1,22 @@
+        c->setProjection(cam->getViewProjection());
+
+        const Style* s = getDefaultStyle();
+        RectangularPrism* rec = new RectangularPrism(c, s, 3, 3, 3, -10, -5, 0);
```
Unfortunately something with either the camera or with the prism itself is not working, as I am not seeing a rectangular prism when I run this.

ahuston-0 commented on test/testTransform.cc:
```diff
@@ -0,0 +1,28 @@
+    const Style* s = getDefaultStyle();
+    Camera* cam = c->setLookAtProjection(0, 0, 30, 0, 0, 0, 0, 1, 0);
+    c->setProjection(cam->getViewProjection());
+    RectangularPrism* rec = new RectangularPrism(c, s, 40, 756, 3, -10, -5, 0);
```
The rectangular prism does show up here, but when I alt-tab or change the window size, it draws something else too. Very strange but kinda cool.

ahuston-0 commented on src/opengl/Sphere.hh:
```diff
@@ -7,8 +7,7 @@ private:
   void addRect(uint32_t i1, uint32_t i2, uint32_t i3, uint32_t i4);
   void addTri(uint32_t i1, uint32_t i2, uint32_t i3);
-  void addVert(float x, float y, float z); // add just coords
-  void addVert(float x, float y, float z, float u, float v); // add with texture
```
why get rid of the textured addVert?

This does not appear to break Globe or SolarSystem, which are the only 3D demos I can think of. So we may be fine to keep this change

ahuston-0 commented on src/opengl/GLWin.cc:
```diff
@@ -335,6 +335,10 @@ GLWin::~GLWin() {
+  //TODO: baseInit() has to be called twice, fix it
+  //seems stupid but called baseInit twice since shapes wouldn't render if it was after, 
+  //but transform in CAD would segfault if it wasn't before
+  baseInit();  // call grails initialization for shaders
```
I'll test the things that get called in baseInit but I don't think this is acceptable for use in the rest of the project.

ahuston-0 replied:
After removing the first baseInit, it looks like its still working. Going to check in with @kaiqichee, but I think this can be removed.

ahuston-0 commented:
src/CAD/Extrude.{cc,hh} are both empty. Both should be removed.

ahuston-0 commented on src/CAD/Prism.hh:
```diff
@@ -0,0 +1,16 @@
+#pragma once
+#include "opengl/Shape.hh"
+#incldue "opengl/Polygon.hh"
```
I don't think this is being used in the project given the misspelled include, so I might end up removing this file.

ahuston-0 commented on src/CAD/Main.cc:
```diff
@@ -0,0 +1,22 @@
+
+void grailmain() {
+  Canvas* c = ...;
+	StevensCAD* cad = c->addLayer(new StevensCAD());
```
A main.cc already exists in another branch that will be merged in. We'll have to check if any conflicts will exist between them.

ahuston-0 replied:
I missed line 7 not being a line that will compile. I will keep this in here as a reference for future projects.

ahuston-0 committed:
Comments out Camera

The camera experiments are cool, however the camera is pointing away from the shapes. We should probably do some more stuff with it later though.

ahuston-0 committed:
Fix Helix import

The test was previously importing HelixPath.hh, but that file does not exist. The test works when using Helix.hh instead.

ahuston-0 committed:
Remove double baseInit

Fixes double baseInit issue

ahuston-0 committed:
Undo textured coord removal
